src/main/webapp/go.js

```diff
@@ -49,24 +58,34 @@ function getHunt() {
  * @param {int} index The index of the destination that the user needs
  * to find.
  */
-function updateToCurrentState(index) {
+async function updateToCurrentState(index) {
+  if (index == 0) { // User presses the start button.
+    sendIndexToServlet(index);
+  }
   if (index >= 0) {
     hideStartButton();
-    changeRiddleMessage(riddleArr[index]);
-    getDestName();
+    changeRiddleMessage(puzzleArr[index]);
+    const isCorrect = await checkCorrectDestination();
+    if (isCorrect) {
+      toggleProceedButton(/* hide = */ false);
+    }
+  } else {
```

Comment:
Is this meant to be an `else` branch that goes with the `if (isCorrect)` statement?

Reply:
Currently, nothing happens when the user enters an incorrect answer, though in the future I'll likely add a message telling the user if their answer was correct or incorrect.

Reply:
Oh, OK, so I guess this `else` is in the place it's meant to be, because it is executed in the case that index is negative, which indicates that the scavenger hunt hasn't started yet, so we hide the proceed button. Right? Could you add a comment about that here?